test/extended/etcd/vertical_scaling.go

```diff
@@ -293,4 +296,68 @@ var _ = g.Describe("[sig-etcd][Feature:EtcdVerticalScaling][Suite:openshift/etcd
 		err = scalingtestinglibrary.EnsureCPMSReplicasConverged(ctx, cpmsClient)
 		o.Expect(err).ToNot(o.HaveOccurred())
 	})
+
+	// The following test covers a vertical scaling scenario when kubelet is not working on a node.
+	// This test validates that deleting the machine hosting the node where the kubelet is stopped doesn't get stuck when CPMS is enabled.
+	//
+	// CPMS should be active for this test scenario
+	// 1) Stop the kubelet on a node
+	// 2) Delete the machine hosting the node in step 2.
+	// 3) That should prompt the ControlPlaneMachineSetOperator(CPMSO) to create a replacement
+	//		machine and node for that machine index
+	// 4) The operator will first scale-up the new machine's member
+	// 5) Then scale-down the machine that is pending deletion by removing its member and deletion hook
+	g.It("is able to vertically scale up and down when kubelet is not running on a node[apigroup:machine.openshift.io][Disruptive]", func() {
+		etcdTargetNode := scalingtestinglibrary.MasterNodes(ctx, kubeClient)[0]
+
+		// step 1: stop the kubelet on a node
+		framework.Logf("Stopping the kubelet on the node %s", etcdTargetNode.Name)
+		err = scalingtestinglibrary.StopKubelet(ctx, oc.AdminKubeClient(), *etcdTargetNode)
```

Comment:
avoid the pointer deref here, just pass it down into the function and error out if the node is nil